docs/DeveloperGuide.md

```diff
@@ -1,29 +1,265 @@
+> - The `XYZ` in `XYZMiniGame` represents the exact mini-game e.g., `TypingGame`, `TicTacToe`, etc.
+> - The `XYZ` in `StockXYZ` represents the stock label e.g., `StockOne`, `StockTwo`, etc.
+
+Here is the sequence diagram of the `EconoCraftLogic` executing the command:
+
+![Work.png](UML%20diagram%2FWork.png)
+
+The `EconoCraftLogic` mechanism:
+
+1. `EconoCraftLogic` receives the user input string and pass it into `CommandFactory`.
+2. `CommandFactory` would use `Parser` to parse the user input string and produce a `Command`.
+3. `EconoCraftLogic` would execute the `Command` which will:
+   * update the player profile accordingly.
+   * use `ResponseManager` to generate the response to the user.
+
+Here is the sequence diagram of `stock` command:
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+![Stock.jpeg](UML%20diagram%2FStock.jpeg)
 
+The mechanism:
+1. `stock` command invoke the `StockCommand` class.
+2. `StockCommand` invokes the `start` method for `StockActivate` class.
+3. `StockActivate` class calls a new `StockStorage` class.
+4. `StockStorage` class will new execute its play function and enable user to buy stocks.
+
+Here is the sequence diagram of `sellstock` command:
+
+![SellStock.jpeg](UML%20diagram%2FSellStock.jpeg)
+
+The mechanism:
+1. `sellstock` command invoke the `SellStockCommand` class.
+2. `SellStockCommand` invokes the `getAsset` method for `PlayerProfile` and fetch the `Asset` class.
+3. `Asset` class executes the `sellStock()` function and sell all of the stocks the player currently possess.
+4. Information related to stocks will be updated inside `Asset` and `PlayerProfile` class.
+
+![Save.png](UML%20diagram%2FSave.png)
+
+The mechanism:
+1. `Saver` will be called automatically after one command is executed.
+2. `Saver` creates a new `file` class and invokes the `constructJson` method for `PlayerProfile` class.
+3. `Saver` class calls `write` and `flush` method from the `file` class to write to the json file.
+
+![Load.png](UML%20diagram%2FLoad.png)
+
+The mechanism:
+1. `Loader` will be called automatically once the program is initialized.
+2. `Loader` invokes the `readJsonFromFile` method to read data from the json file and creates a new `file` class.
+3. `Loader` invokes the `parseJson` method to retrieve relevant attributes and creates a new `PlayerProfile` class.
+4. `Loader` returns the `PlayerProfile` class to load the previous information.
+
+
+## MiniGame components
+
+Here is the partial class diagram of the `MiniGame` component:
+
+![MiniGame.png](UML%20diagram%2FMiniGame.png)
+
+The `MiniGame` mechanism:
+1. For command `work`, `rest`, and `exercise`, they would have their respective mini-games.
+2. When these commands are generated and executed in `EconoCraftLogic`, the respective mini-game would be played.
+3. The command would then update the player profile according to the mini-game result.
+
+# Implementation
+
```

Comment:
There seems to be enough examples e.g., sample inputs/outputs. i can understand.